### ISSUE TITLE
When using https as a URL in a callback the callback does not appear to be called

### ISSUE THREAD

**Issue by simonhitachi** [mockoon/mockoon#1273]
### Discussed in https://github.com/mockoon/mockoon/discussions/1272

<div type='discussions-op-text'>

<sup>Originally posted by **simonhitachi** January 19, 2024</sup>
I have set up some mocking with callbacks on some of the routes as the callbacks reference the same mock server they initially worked when using http. When I changed this to https the callback did not appear to be called (there was no reference to this in the Mockoon UI logs). I even tried pointing this to an external API service using HTTPS and had the same behaviour. One point of note, the endpoint is self-signed.

Is it actually possibly to use a HTTPS callback? When creating a new callback the example URL shown is https, but currently can't get anything other than http requests to get triggered via  a callback.

From the discussion response I've had this is because self-signed certificates are not currently supported in callbacks. As Mockoon is a dev/testing tool, it makes sense to allow self-signed certs.

Running Mockoon v6.1.0 on Windows 10.</div>

**Comment by isuru89** [comment]
I agree that this is a very common scenario in dev/test environments.

However, for now, the native library we use for making callbacks is the fetch API, and it has very limited configurations to make it work with TLS setups.
